automation/WinFormsAutomationApp/AuthenticationHelper.cs

```diff
@@ -230,5 +246,61 @@ private static IPlatformParameters GetPlatformParametersInstance(string promptBe
 #endif
             return platformParameters;
         }
+
+        public static string FromDictionaryToJson(this Dictionary<string, string> dictionary)
+        {
+            var kvs = dictionary.Select(kvp => string.Format("\"{0}\":\"{1}\"", kvp.Key, string.Concat(kvp.Value)));
+            return string.Concat("{", string.Join(",", kvs), "}");
+        }
+
+        private static Dictionary<string, string> ProcessResult(AuthenticationResult result, Dictionary<string, string> input)
+        {
+            Dictionary<string, string> res = new Dictionary<string, string>();
+            res.Add("unique_id", result.UserInfo.UniqueId);
+            res.Add("access_token", result.AccessToken);
+            res.Add("tenant_id", result.TenantId);
+            res.Add("refresh_token", TokenCache.DefaultShared.tokenCacheDictionary[new TokenCacheKey(input["authority"], input["resource"], input["client_id"], TokenSubjectType.User, result.UserInfo)].RefreshToken);
+            return res;
+        }
+
+        private static void NotifyBeforeAccessCache(string resource, string clientid, string uniqueid, string displayableid)
+        {
+            TokenCache.DefaultShared.OnBeforeAccess(new TokenCacheNotificationArgs
+            {
+                TokenCache = TokenCache.DefaultShared,
+                Resource = resource,
+                ClientId = clientid,
+                UniqueId = uniqueid,
+                DisplayableId = displayableid
+            });
+        }
+
+        private static void NotifyAfterAccessCache(string resource, string clientid, string uniqueid, string displayableid)
+        {
+            TokenCache.DefaultShared.OnAfterAccess(new TokenCacheNotificationArgs
+            {
+                TokenCache = TokenCache.DefaultShared,
+                Resource = resource,
+                ClientId = clientid,
+                UniqueId = uniqueid,
+                DisplayableId = displayableid
+            });
+        }
+
+        private static void UpdateCache(KeyValuePair<TokenCacheKey, AuthenticationResultEx> item, KeyValuePair<TokenCacheKey, AuthenticationResultEx> updated)
+        {
+            bool notifiedBeforeAccessCache = false;
```

Comment:
This flag does not seem to be needed. It's simply set to true and the if branch will always execute.